EIPS/eip-5593.md

```diff
@@ -0,0 +1,83 @@
+    - Secure contexts include sites that are served from HTTPS but also HTTP `localhost`.
+- By default the web3 provider object APIs MUST NOT be exposed to 3p iframes.
+- `window.ethereum` MUST be `undefined` in an iframe where `window.isSecureContext` returns `false` in that iframe.
+- If the iframe is first-party to the top-level origin AND the `sandbox` attribute is set on the iframe, the provider object MUST be left `undefined`. If `sandbox="allow-same-origin"` attribute is set the provider object MAY be injected. 
+    - Note `"allow-same-origin"` does nothing if the iframe is third-party.
+- If the iframe is a third party to the top-level secure origin, it SHOULD be blocked. It MAY be unblocked if  the iframe uses the `allow=ethereum` attribute. In order to achieve this implementers are REQUIRED to extend the Permissions API.
+
+## Rationale
+
+By limiting the capabilities of where the web3 provider object APIs are being injected we can reduce the surface area of where attacks can be executed.Given the sensitivity of data that's passed to the web3 provider object APIs some basic levels of authentication and confidentiality should be met in order to ensure that request data is not being intercepted or tampered with. While there has been attempts to [limit request access via the wallet](./eip-2255.md) interface itself there's not been limitations that have been set to where these web3 provider object APIs are expected to be or not be injected. Since the secure contexts web platform API is a well developed boundary that's been recommended by W3C and the fact that the web3 provider object APIs are extending the traditional web platform APIs, no other alternative solutions have been considered in order to extend current established prior art.
+
+
+## Backwards Compatibility
+
+Wallet extensions SHOULD consider adding a "developer mode" toggle via a UX so that dApp developers have the capability to disable the insecure context (http) check for the http://localhost:<any-port> origin only in the event that [localhost does not return `true` for secure context](https://www.w3.org/TR/secure-contexts/#localhost). This will allow dApp developers to be able to continue to host dApps on the localhost origin if a browser environment has chosen to not already consider localhost a secure context. Most major browser providers do consider localhost a secure context already. This toggle MUST be set to disabled by default.
+
+## Test Cases
+
+### Required Test Cases
+
+- Top level `http://a.com` -> blocked (insecure)
+- Top level `https://a.com` -> allowed
+- Top level `https://a.com` with `<iframe src="http://a.com/">` -> blocked (insecure/3p)
+- Top level `http://a.com` with `<iframe src="https://a.com/">` -> blocked (insecure/3p)
```

Comment:
insure first party right?

Reply:
Yeah that's insecure top level not 3p. Will get this switched. Good catch!